lib/Onyx.js

```diff
@@ -931,14 +960,9 @@ function applyMerge(changes, existingValue) {
 
     if (_.isObject(existingValue) || _.every(changes, _.isObject)) {
         // Object values are merged one after the other
-        return _.reduce(changes, (modifiedData, change) => {
-            // lodash adds a small overhead so we don't use it here
-            // eslint-disable-next-line prefer-object-spread, rulesdir/prefer-underscore-method
-            const newData = Object.assign({}, fastMerge(modifiedData, change));
-
-            // Remove all first level keys that are explicitly set to null.
-            return _.omit(newData, value => _.isNull(value));
-        }, existingValue || {});
+        // lodash adds a small overhead so we don't use it here
```

Comment:
This comment is no longer valid. It also makes me think that we should still be using `Object.assign()` instead of the spread operator since performance is a very big consideration in this code.

Reply:
Oh i thought the lodash comment was referring to that we the `fastMerge` instead of `_.merge()`.

According to [this source](https://thecodebarbarian.com/object-assign-vs-object-spread.html), object spreading seems to be the faster choice, especially when we pass an empty object as the target to `Object.assign`, as we do right now.

Reply:
I'd be careful referring to that article which is about Node.js. I've done a lot of performance analyzing in the browser and found the spread operator to be much slower than `Object.assign` (which could have just as much to do with our transpiler).

Reply:
Alright! Let's switch back to `Object.assign` then. 👍

Reply:
Technically, we don't need the `Object.assign` after all, because fast merge is already copying into a new object:

https://github.com/Expensify/react-native-onyx/blob/d4fc53cc3229808e9c1c542c13ce0089e5893d96/lib/fastMerge.js#L20

Reply:
I'm all for removing it entirely!